ui/src/components/EventConfiguration/eventConfigXmlValidator.ts

```diff
@@ -0,0 +1,90 @@
+export const validateEventConfigFile = async (file: File) => {
+  const validationErrors: string[] = []
+
+  try {
+    const text = await file.text()
+
+    if (text.trim().length === 0) {
+      validationErrors.push('File is empty')
+      return { isValid: false, errors: validationErrors }
+    }
+
+    const parser = new DOMParser()
+    const xmlDoc = parser.parseFromString(text, 'application/xml')
+    
+    if (xmlDoc.querySelector('parsererror')) {
+      validationErrors.push('Invalid XML format - file contains syntax errors')
+      return { isValid: false, errors: validationErrors }
+    }
+
+    const eventsElement = xmlDoc.querySelector('events')
+    if (!eventsElement) {
+      validationErrors.push('Missing <events> root element')
+    } else {
+      const xmlns = eventsElement.getAttribute('xmlns') || ''
+      if (!xmlns.includes('opennms.org')) {
+        validationErrors.push('Missing or invalid OpenNMS namespace in <events> element')
+      }
+    }
+
+    const eventElements = xmlDoc.querySelectorAll('event')
+    if (eventElements.length === 0) {
+      validationErrors.push('No <event> entries found within <events> element')
+      
+      if (eventsElement && eventElements.length === 0) {
+        const childElements = eventsElement.children
+        if (childElements.length === 0) {
+          validationErrors.push('Empty <events> element - no content found')
+        } else {
+          const childNames = Array.from(childElements).map(el => `<${el.tagName}>`).join(', ')
+          validationErrors.push(`<events> element contains ${childNames} but no <event> elements`)
+        }
+      }
+    } else {
+      eventElements.forEach((event, idx) => {
+        const eventErrors = validateEventElement(event, idx + 1)
+        validationErrors.push(...eventErrors)
```

Comment:
Should only push `...eventErrors` if it is non-empty, otherwise it will add an empty array and later on `isValid: validationErrors.length === 0` will be incorrect.